juju/juju
# Validate LXDProfile for charm store/repo

## Description of change

The following changes validates the LXDProfile of a charm and bundle
if it's from the charm store.

The following checks that the profile is correctly validated when deploying.

## Notes

Feedback for where the `validateXXX` profile function should be, it's currently
duplicated and would make sense to have it somewhere for reuse.

## QA steps

Ensure that you have the feature flag turned on when bootstrapping, as 
this needs to be set.

```sh
export JUJU_DEV_FEATURE_FLAGS=lxd-profile
```

The cases I checked are the following, I'm sure I probably missed one,
so if that's the case let me know so I can ensure that we cover those
cases as well.

```sh
juju deploy ./testcharms/charm-repo/bundle/lxd-profile/bundle.yaml
juju deploy ./testcharms/charm-repo/bundle/lxd-profile-fail/bundle.yaml
juju deploy cs:~juju-qa/bionic/lxd-profile-0
juju deploy cs:~juju-qa/bionic/lxd-profile-fail-0
```

## Documentation changes

When deploying a bundle with a lxd-profile configuration that isn't valid,
it should show an error to the user explaining what went wrong and 
potentially how to fix it.

## Bug reference

TBA
